crates/wasmparser/src/readers/core/types.rs

```diff
@@ -1454,53 +1553,14 @@ impl<'a> FromReader<'a> for RefType {
 impl<'a> FromReader<'a> for HeapType {
     fn from_reader(reader: &mut BinaryReader<'a>) -> Result<Self> {
         match reader.peek()? {
-            0x70 => {
+            0x65 => {
                 reader.read_u8()?;
-                Ok(HeapType::Func)
+                let ty = reader.read()?;
+                Ok(HeapType::Abstract { shared: true, ty })
             }
-            0x6F => {
-                reader.read_u8()?;
-                Ok(HeapType::Extern)
-            }
-            0x6E => {
-                reader.read_u8()?;
-                Ok(HeapType::Any)
-            }
-            0x71 => {
-                reader.read_u8()?;
-                Ok(HeapType::None)
-            }
-            0x72 => {
-                reader.read_u8()?;
-                Ok(HeapType::NoExtern)
-            }
-            0x73 => {
-                reader.read_u8()?;
-                Ok(HeapType::NoFunc)
-            }
-            0x6D => {
-                reader.read_u8()?;
-                Ok(HeapType::Eq)
-            }
-            0x6B => {
-                reader.read_u8()?;
-                Ok(HeapType::Struct)
-            }
-            0x6A => {
-                reader.read_u8()?;
-                Ok(HeapType::Array)
-            }
-            0x6C => {
-                reader.read_u8()?;
-                Ok(HeapType::I31)
-            }
-            0x69 => {
-                reader.read_u8()?;
-                Ok(HeapType::Exn)
-            }
-            0x74 => {
-                reader.read_u8()?;
-                Ok(HeapType::NoExn)
+            0x70 | 0x6F | 0x6E | 0x71 | 0x72 | 0x73 | 0x6D | 0x6B | 0x6A | 0x6C | 0x69 | 0x74 => {
```

Comment:
This reminds me that I've been meaning for some time now to clean up all the decoding here in this crate since all the various constants all over the place feel a bit overkill.

Reply:
Looks like that happened in #1601; resolving this.